app/persistence/transfer/transfer.go

```diff
@@ -135,6 +137,54 @@ func (r *Repository) UpdateStatusFailed(txId string) error {
 	return r.updateStatus(txId, status.Failed)
 }
 
+func formatTimestampFilter(q *gorm.DB, ts_query string) (*gorm.DB, error) {
+	q_params := strings.Split(ts_query, "&")
+
+	// This if statement handles legacy timestamp filter like:
+	// "timestamp": "2023-04-19T04:41:47.104114905Z"
+	if len(q_params) == 1 && !strings.Contains(ts_query, "=") {
+		timestamp, err := time.Parse(time.RFC3339Nano, q_params[0])
+		if err != nil {
+			return q, err
+		}
+
+		q = q.Where("timestamp = ?", timestamp.UnixNano())
+		return q, nil
+	}
+
+	// This for loop handles new timestamp filter like:
+	// "timestamp": "lte=2023-05-19T04:41:47.104114905Z&gte=2023-04-19T04:41:47.104114905Z"
+	// "timestamp": "eq=2023-04-19T04:41:47.104114905Z"
+	for _, param := range q_params {
+		parts := strings.Split(param, "=")
+
+		operator := parts[0]
+		datetime, err := time.Parse(time.RFC3339Nano, parts[1])
+		if err != nil {
+			return q, err
+		}
+
+		timestamp := datetime.UnixNano()
+
+		switch operator {
+		case "eq":
+			q = q.Where("timestamp = ?", timestamp)
+		case "gt":
+			q = q.Where("timestamp > ?", timestamp)
+		case "lt":
+			q = q.Where("timestamp < ?", timestamp)
+		case "gte":
+			q = q.Where("timestamp >= ?", timestamp)
+		case "lte":
+			q = q.Where("timestamp <= ?", timestamp)
+		default:
+			return q, fmt.Errorf("invalid operator: %s", operator)
+		}
```

Comment:
WDYT about this approach:

```
operators := map[string]string{
	"eq":  "=",
	"gt":  ">",
	"lt":  "<",
	"gte": ">=",
	"lte": "<=",
}

op, exists := operators[operator]
if !exists {
	return q, fmt.Errorf("invalid operator: %s", operator)
}

q = q.Where("timestamp "+op+" ?", timestamp)
```

with the map of operators being defined as a constant somewhere on top

Reply:
Thats a badass solution. I never tough about hash map in such a way. I like it!